deadpool.py

```diff
@@ -137,7 +137,7 @@ def runner(self):
             if job is None:
                 # This is for the `None` that terminates the while loop.
                 self.submitted_jobs.task_done()
-                self.running_jobs.get()
+                self.running_jobs.get_nowait()
```

Comment:
I think this change is wrong. The intention is to do a blocking wait on the `get()` call.